turbinia/turbiniactl.py

```diff
@@ -854,40 +856,28 @@ def main():
     request = TurbiniaRequest(
         request_id=request_id, requester=getpass.getuser())
     request.evidence.append(evidence_)
-    if filter_patterns:
-      request.recipe['filter_patterns'] = filter_patterns
-    if args.jobs_denylist:
-      request.recipe['jobs_denylist'] = args.jobs_denylist
-    if args.jobs_allowlist:
-      request.recipe['jobs_allowlist'] = args.jobs_allowlist
-    if yara_rules:
-      request.recipe['yara_rules'] = yara_rules
-    if args.debug_tasks:
-      request.recipe['debug_tasks'] = args.debug_tasks
-    if args.recipe_config:
-      for pair in args.recipe_config:
-        try:
-          key, value = pair.split('=')
-        except ValueError as exception:
-          log.error(
-              'Could not parse key=value pair [{0:s}] from recipe config '
-              '{1!s}: {2!s}'.format(pair, args.recipe_config, exception))
-          sys.exit(1)
-        request.recipe[key] = value
-    if args.decryption_keys:
-      for credential in args.decryption_keys:
-        try:
-          credential_type, credential_data = credential.split('=')
-        except ValueError as exception:
-          log.error(
-              'Could not parse credential [{0:s}] from decryption keys '
-              '{1!s}: {2!s}'.format(
-                  credential, args.decryption_keys, exception))
-          sys.exit(1)
-        evidence_.credentials.append({
-            'credential_type': credential_type,
-            'credential_data': credential_data
-        })
+
+    if args.recipe:
+      if (args.jobs_denylist or args.jobs_allowlist or
+          args.filter_patterns_file or args.yara_rules_file):
+        raise TurbiniaException(
+            'Specifying a recipe is incompatible with defining'
+            ' jobs allow/deny lists, yara rules or a patterns file separately.')
+        sys.exit(1)
+      recipe_file = os.path.join(config.RECIPE_FILE_DIR, args.recipe)
+      recipe_dict = recipe_helpers.load_recipe_from_file(recipe_file)
+      if not recipe_dict:
+        sys.exit(1)
```

Comment:
The load recipe function within the recipe helpers already throws a meaningful error that the user launching turbiniactl will be able to see. If we include something it could be something to the effect of "Terminating now..", but probably enough as it is.